### PR TITLE
[5.0] TinyMCE use assets for its plugins

### PR DESCRIPTION
**Pull request by Fedik**
Pull Request for Issue # .

### Summary of Changes
This update for TinyMCE to use asset registry, 
And regular plugin, instead of  external_plugin and  `HTMLHelper::_('script', 'plg_ .....` for Joomla custom plugins ,
Also this create `jxtdbuttons` plugin, to render XTD buttons, instead of hardocing it.

### Testing Instructions
Apply patch, run `npm install`
Edit an article with TineMCE, also use  XTD buttons

### Actual result BEFORE applying this Pull Request
Works

### Expected result AFTER applying this Pull Request
Works, but better

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [x] No documentation changes for docs.joomla.org needed
- [ ] Pull Request link for manual.joomla.org: <link>
- [x] No documentation changes for manual.joomla.org needed

### CODE REVIEW

File: build/media_source/plg_editors_tinymce/js/plugins/jxtdbuttons/plugin.es6.js [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    text: Joomla.Text._('PLG_TINY_CORE_BUTTONS'),
+    icon: 'joomla',
+    fetch: (callback) => callback(subMenu),
+  });
+};
+
+window.tinymce.PluginManager.add('jxtdbuttons', (editor) => {
+  pluginSetUp(editor);
+  return {
+    getMetadata: () => ({
```
Comment (brianteeman) [inline build/media_source/plg_editors_tinymce/js/plugins/jxtdbuttons/plugin.es6.js:68]: ;)

thank you!!

**Comment by brianteeman** [comment]
Before this PR If you disable all of the plugins of type editors-xtd then there is no CMS Content button
With this PR you still have the button even though it is empty

**Comment by Fedik** [comment]
> With this PR you still have the button even though it is empty

Yeap, should work now.

**Comment by brianteeman** [comment]
I have tested this item :white_check_mark: successfully on 0635294c09000bc31f0f4439f9f76a72cd5fbed8<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/41595">issues.joomla.org/tracker/joomla-cms/41595</a>.</sub>

**Comment by ceford** [comment]
I have tested this item :white_check_mark: successfully on d720009726c3cf84b9b594e0c1b825653176a315<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/41595">issues.joomla.org/tracker/joomla-cms/41595</a>.</sub>

**Comment by HLeithner** [comment]
thanks

**Pull request merged**